openedx/openedx-platform
# refactor: inheritable authoring mixin callbacks for editing & duplication

## Description

Instead of applying duplicated logic after getattr, that logic is incorporated into inheritable methods of the StudioEditable block. Should not affect other roles besides developers.

## Supporting information

Resolves https://github.com/openedx/edx-platform/issues/33715

## Testing instructions

Courses:
- Log into CMS and go to Any Course
- Duplicate any Section
- Make sure the Section is duplicated with all its children

Libraries:
- Navigate to the libraries tab
- Create a new library
- Try adding, editing and duplicating a text, a video, and a problem component
- Make sure all student previews are correctly displayed


## Deadline

None
